Mathlib/Probability/Moments.lean

```diff
@@ -285,6 +286,28 @@ theorem iIndepFun.cgf_sum {X : ι → Ω → ℝ}
   · rw [h_indep.mgf_sum h_meas]
   · exact (mgf_pos (h_int j hj)).ne'
 
+theorem mgf_ident_distrib
+    (X : Ω → ℝ) {Ω' : Type u_1} {m' : MeasurableSpace Ω'} {μ' : Measure Ω'} (X' : Ω' → ℝ)
+    (hident : IdentDistrib X X' μ μ') (t : ℝ) :
+    mgf X μ t = mgf X' μ' t := by
+  rw [mgf, mgf]
+  apply IdentDistrib.integral_eq
+  let u := fun x => Real.exp (t * x)
+  have compX : (fun ω ↦ Real.exp (t * X ω)) = u ∘ X := rfl
+  have compX' : (fun ω ↦ Real.exp (t * X' ω)) = u ∘ X' := rfl
+  rw [compX, compX']
+  exact IdentDistrib.comp hident (Measurable.exp (measurable_const_mul t))
```

Comment:
You're telling Lean to rewrite things that it already knows are the same (as you prove them by `rfl`). This means that this is not really necessary, and that you can just skip all these steps, to get:
```suggestion
  apply IdentDistrib.integral_eq
  apply IdentDistrib.comp hident (Measurable.exp (measurable_const_mul t))
```